diff --git a/src/main/scala/v3/common/parameters.scala b/src/main/scala/v3/common/parameters.scala
index 2f5e1268..fef38a71 100644
--- a/src/main/scala/v3/common/parameters.scala
+++ b/src/main/scala/v3/common/parameters.scala
@@ -100,7 +100,7 @@ case class BoomCoreParams(
   trace: Boolean = false,
 
   /* debug stuff */
-  enableCommitLogPrintf: Boolean = false,
+  enableCommitLogPrintf: Boolean = true,
   enableBranchPrintf: Boolean = false,
   enableMemtracePrintf: Boolean = false
 
diff --git a/src/main/scala/v4/common/parameters.scala b/src/main/scala/v4/common/parameters.scala
index 0e079fc9..710b6b37 100644
--- a/src/main/scala/v4/common/parameters.scala
+++ b/src/main/scala/v4/common/parameters.scala
@@ -117,7 +117,7 @@ case class BoomCoreParams(
   trace: Boolean = false,
 
   /* debug stuff */
-  enableCommitLogPrintf: Boolean = false,
+  enableCommitLogPrintf: Boolean = true,
   enableBranchPrintf: Boolean = false,
   enableMemtracePrintf: Boolean = false
 
