diff --git a/Makefile b/Makefile
index f2608a3..9944cd2 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ ifeq ($(shell echo $(GCC_VERSION)\>=9 | bc ),0)
   SCL_PREFIX := source scl_source enable devtoolset-10 &&
 endif
 
-all: bitstream
+all: vlt
 
 
 #######################################
@@ -45,7 +45,7 @@ endif
 #######################################
 
 ROCKET_TOP		:= $(STARSHIP_TH)
-ROCKET_CONF		:= starship.With$(STARSHIP_CORE)Core,$(STARSHIP_CONFIG),starship.With$(STARSHIP_FREQ)MHz
+ROCKET_CONF		:= starship.With$(STARSHIP_CORE)Core,$(STARSHIP_CONFIG) #,starship.With$(STARSHIP_FREQ)MHz
 ROCKET_SRC		:= $(SRC)/rocket-chip
 ROCKET_BUILD	:= $(BUILD)/rocket-chip
 ROCKET_SRCS     := $(shell find $(TOP) -name "*.scala")
@@ -146,29 +146,6 @@ verilog-patch: verilog
 	sed -i "s/_covSum = _RAND/_covSum = 0; \/\//g" $(ROCKET_TOP_VERILOG)
 
 
-#######################################
-#
-#         Bitstream Generator
-#
-#######################################
-
-VIVADO_TOP			:= $(lastword $(subst ., ,$(STARSHIP_TH)))
-VIVADO_SRC			:= $(SRC)/rocket-chip-fpga-shells
-VIVADO_SCRIPT		:= $(VIVADO_SRC)/xilinx
-VIVADO_BUILD		:= $(BUILD)/vivado
-VIVADO_BITSTREAM 	:= $(VIVADO_BUILD)/$(ROCKET_OUTPUT).bit
-
-$(VIVADO_BITSTREAM): $(ROCKET_INCLUDE) $(VERILOG_SRC)
-	mkdir -p $(VIVADO_BUILD)
-	cd $(VIVADO_BUILD); vivado -mode batch -nojournal \
-		-source $(VIVADO_SCRIPT)/common/tcl/vivado.tcl \
-		-tclargs -F "$(ROCKET_INCLUDE)" \
-		-top-module "$(VIVADO_TOP)" \
-		-ip-vivado-tcls "$(shell find '$(ROCKET_BUILD)' -name '*.vivado.tcl')" \
-		-board "$(STARSHIP_BOARD)"
-
-bitstream: $(VIVADO_BITSTREAM)
-
 #######################################
 #
 #         RTL Simulation
@@ -207,61 +184,6 @@ $(SPIKE_BUILD)/Makefile:
 $(SPIKE_LIB)&: $(SPIKE_SRC) $(SPIKE_BUILD)/Makefile
 	cd $(SPIKE_BUILD); $(SCL_PREFIX) make -j$(shell nproc) $(notdir $(SPIKE_LIB))
 
-
-#######################################
-#
-#            Synopsys VCS
-#
-#######################################
-
-VCS_OUTPUT	:= $(BUILD)/vcs
-VERDI_OUTPUT:= $(BUILD)/verdi
-VCS_BUILD	:= $(VCS_OUTPUT)/build
-VCS_LOG		:= $(VCS_OUTPUT)/log
-VCS_WAVE	:= $(VCS_OUTPUT)/wave
-
-VCS_TARGET	:= $(VCS_BUILD)/$(TB_TOP)
-VCS_INCLUDE	:= $(ROCKET_BUILD)+$(SIM_DIR)
-VCS_CFLAGS	:= -std=c++17 $(addprefix -I,$(SPIKE_INCLUDE)) -I$(ROCKET_BUILD)
-
-VCS_SRC_C	:= $(SIM_DIR)/spike_difftest.cc \
-			   $(SPIKE_LIB) \
-			   $(SIM_DIR)/timer.cc  
-
-VCS_SRC_V	:= $(SIM_DIR)/$(TB_TOP).v \
-			   $(SIM_DIR)/spike_difftest.v \
-			   $(SIM_DIR)/tty.v
-
-VCS_DEFINE	:= +define+MODEL=$(STARSHIP_TH)					\
-			   +define+TOP_DIR=\"$(VCS_OUTPUT)\"			\
-			   +define+INITIALIZE_MEMORY					\
-			   +define+CLOCK_PERIOD=1.0	   					\
-			   +define+DEBUG_FSDB							\
-			   +define+TARGET_$(STARSHIP_CORE)
-
-VCS_PARAL_COM	:= -j$(shell nproc) # -fgp
-VCS_PARAL_RUN	:= # -fgp=num_threads:1,num_fsdb_threads:1 # -fgp=num_cores:$(shell nproc),percent_fsdb_cores:30
-
-VCS_OPTION	:= -quiet -notice -line +rad -full64 +nospecify +notimingcheck -deraceclockdata 		\
-			   -sverilog +systemverilogext+.sva+.pkg+.sv+.SV+.vh+.svh+.svi+ -assert svaext 			\
-			   +vcs+initreg+random +v2k -debug_acc+all -timescale=1ns/10ps +incdir+$(VCS_INCLUDE) 	\
-			   $(VCS_PARAL_COM) -CFLAGS "$(VCS_CFLAGS)" 											\
-			   $(CHISEL_DEFINE) $(VCS_DEFINE)
-VCS_SIM_OPTION	:= +vcs+initreg+random $(VCS_PARAL_RUN) +testcase=$(TESTCASE_ELF)
-
-vcs-wave: 		VCS_SIM_OPTION += +dump +uart_tx=0
-vcs-debug: 		VCS_SIM_OPTION += +verbose +dump +uart_tx=0
-vcs-fuzz: 		VCS_SIM_OPTION += +fuzzing +uart_tx=0
-vcs-fuzz-debug:	VCS_SIM_OPTION += +fuzzing +verbose +dump +uart_tx=0
-vcs-jtag: 		VCS_SIM_OPTION += +jtag_rbb_enable=1 +verbose +uart_tx=0
-vcs-jtag-debug: VCS_SIM_OPTION += +jtag_rbb_enable=1 +verbose +dump +uart_tx=0
-
-$(VCS_TARGET): $(VERILOG_SRC) $(ROCKET_ROM_HEX) $(ROCKET_INCLUDE) $(VCS_SRC_V) $(VCS_SRC_C) $(SPIKE_LIB)
-	$(MAKE) verilog-patch
-	mkdir -p $(VCS_BUILD) $(VCS_LOG) $(VCS_WAVE)
-	cd $(VCS_BUILD); $(SCL_PREFIX) vcs $(VCS_OPTION) -l $(VCS_LOG)/vcs.log -top $(TB_TOP) \
-		-f $(ROCKET_INCLUDE) $(VCS_SRC_V) $(VCS_SRC_C) -o $@
-
 $(TESTCASE_HEX): $(TESTCASE_ELF)
 	riscv64-unknown-elf-objcopy --gap-fill 0			\
 		--set-section-flags .bss=alloc,load,contents	\
@@ -271,24 +193,6 @@ $(TESTCASE_HEX): $(TESTCASE_ELF)
 	od -v -An -tx8 $(TESTCASE_BIN) > $@
 	rm $(TESTCASE_BIN)
 
-vcs: $(VCS_TARGET) $(TESTCASE_HEX)
-	mkdir -p $(VCS_BUILD) $(VCS_LOG) $(VCS_WAVE)
-	cd $(VCS_BUILD); \
-	$(VCS_TARGET) -quiet +ntb_random_seed_automatic -l $(VCS_LOG)/sim.log  \
-		$(VCS_SIM_OPTION) 2>&1 | tee /tmp/rocket.log; exit "$${PIPESTATUS[0]}";
-
-vcs-wave vcs-debug: vcs
-vcs-fuzz vcs-fuzz-debug: vcs
-vcs-jtag vcs-jtag-debug: vcs
-
-verdi:
-	mkdir -p $(VERDI_OUTPUT)
-	touch $(VERDI_OUTPUT)/signal.rc
-	cd $(VERDI_OUTPUT); \
-	verdi -$(VCS_OPTION) -q -ssy -ssv -ssz -autoalias \
-		-ssf $(VCS_WAVE)/starship.fsdb -sswr $(VERDI_OUTPUT)/signal.rc \
-		-logfile $(VCS_LOG)/verdi.log -top $(TB_TOP) -f $(ROCKET_INCLUDE) $(VCS_SRC_V) &
-
 #######################################
 #
 #            Verilator
@@ -343,32 +247,6 @@ vlt-jtag-debug: vlt
 gtkwave:
 	gtkwave $(VLT_WAVE)/starship.vcd
 
-#######################################
-#
-#             Sythesis
-#
-#######################################
-
-#######################################
-#
-#            Synopsys DC
-#
-#######################################
-
-DC_TOP		:= $(lastword $(subst ., ,$(STARSHIP_TH)))
-DC_SRC		:= $(ASIC)/scripts/syn
-DC_OUTPUT	:= $(BUILD)/syn
-DC_BUILD	:= $(DC_OUTPUT)/build
-DC_LOG		:= $(DC_OUTPUT)/log
-DC_NETLIST	:= $(DC_OUTPUT)/netlist
-
-#######################################
-#
-#              Yosys
-#
-#######################################
-
-
 #######################################
 #
 #               Utils
diff --git a/build.sbt b/build.sbt
index c37a0d3..0896942 100644
--- a/build.sbt
+++ b/build.sbt
@@ -56,13 +56,13 @@ lazy val peripheral_blocks = (project in file("repo/rocket-chip-blocks"))
   .dependsOn(rocket_chip, cde)
   .settings(commonSettings)
 
-lazy val fpga_shells = (project in file("repo/rocket-chip-fpga-shells"))
-  .dependsOn(rocket_chip, peripheral_blocks, cde)
-  .settings(
-    commonSettings,
-    Compile / unmanagedBase := baseDirectory.value,
-    Compile / resourceDirectory := baseDirectory.value
-  )
+// lazy val fpga_shells = (project in file("repo/rocket-chip-fpga-shells"))
+//   .dependsOn(rocket_chip, peripheral_blocks, cde)
+//   .settings(
+//     commonSettings,
+//     Compile / unmanagedBase := baseDirectory.value,
+//     Compile / resourceDirectory := baseDirectory.value
+//   )
 
 lazy val ucb_testchipip = (project in file("repo/testchipip/src"))
   .dependsOn(rocket_chip, peripheral_blocks)
@@ -81,7 +81,7 @@ lazy val ucb_boom = (project in file("repo/riscv-boom/src"))
   )
 
 lazy val starship = (project in file("repo/starship"))
-  .dependsOn(rocket_chip, cde, peripheral_blocks, fpga_shells, ucb_boom)
+  .dependsOn(rocket_chip, cde, peripheral_blocks, ucb_boom) // fpga_shells, 
   .settings(commonSettings)
 
 lazy val root = (project in file("."))
diff --git a/conf/build.mk b/conf/build.mk
index 7cb7c8f..44d1096 100644
--- a/conf/build.mk
+++ b/conf/build.mk
@@ -3,9 +3,9 @@
 
 STARSHIP_CORE	?= Rocket
 STARSHIP_FREQ	?= 100
-STARSHIP_TH 	?= starship.fpga.TestHarness
-STARSHIP_TOP	?= starship.fpga.StarshipFPGATop
-STARSHIP_CONFIG	?= starship.fpga.StarshipFPGAConfig
+STARSHIP_TH = starship.asic.TestHarness
+STARSHIP_TOP = starship.asic.StarshipSimTop
+STARSHIP_CONFIG = starship.asic.StarshipSimConfig
 
 
 # FPGA Configuration
diff --git a/repo/riscv-boom b/repo/riscv-boom
--- a/repo/riscv-boom
+++ b/repo/riscv-boom
@@ -1 +1 @@
-Subproject commit 22bb6f420c52f22c3bd94653a2e9813874b45f10
+Subproject commit 22bb6f420c52f22c3bd94653a2e9813874b45f10-dirty
diff --git a/repo/rocket-chip b/repo/rocket-chip
--- a/repo/rocket-chip
+++ b/repo/rocket-chip
@@ -1 +1 @@
-Subproject commit e3773366a5c473b6b45107f037e3130f4d667238
+Subproject commit e3773366a5c473b6b45107f037e3130f4d667238-dirty
diff --git a/repo/rocket-chip-fpga-shells b/repo/rocket-chip-fpga-shells
--- a/repo/rocket-chip-fpga-shells
+++ b/repo/rocket-chip-fpga-shells
@@ -1 +1 @@
-Subproject commit 2ce3e6f3df06d64c858bc1073ba1c75e7eb71a07
+Subproject commit 2ce3e6f3df06d64c858bc1073ba1c75e7eb71a07-dirty
diff --git a/repo/starship/src/main/scala/Configs.scala b/repo/starship/src/main/scala/Configs.scala
index 7f0ef39..f4840b7 100644
--- a/repo/starship/src/main/scala/Configs.scala
+++ b/repo/starship/src/main/scala/Configs.scala
@@ -1,6 +1,6 @@
 package starship
 
-import starship.fpga._
+// import starship.fpga._
 
 import chisel3._
 
@@ -11,27 +11,27 @@ import freechips.rocketchip.diplomacy._
 import freechips.rocketchip.devices.debug._
 import freechips.rocketchip.devices.tilelink._
 
-import sifive.fpgashells.shell._
-import sifive.fpgashells.clocks._
-import sifive.fpgashells.ip.xilinx._
-import sifive.fpgashells.shell.xilinx._
-import sifive.fpgashells.devices.xilinx.xilinxvc707mig._
+// import sifive.fpgashells.shell._
+// import sifive.fpgashells.clocks._
+// import sifive.fpgashells.ip.xilinx._
+// import sifive.fpgashells.shell.xilinx._
+// import sifive.fpgashells.devices.xilinx.xilinxvc707mig._
 
-import sifive.blocks.devices.uart._
-import sifive.blocks.devices.spi._
+// import sifive.blocks.devices.uart._
+// import sifive.blocks.devices.spi._
 
 import sys.process._
 
 case object FrequencyKey extends Field[Double](50)   // 50 MHz
 
-class WithFrequency(MHz: Double) extends Config((site, here, up) => {
-  case FrequencyKey => MHz
-})
+// class WithFrequency(MHz: Double) extends Config((site, here, up) => {
+//   case FrequencyKey => MHz
+// })
 
-class With25MHz  extends WithFrequency(25)
-class With50MHz  extends WithFrequency(50)
-class With100MHz extends WithFrequency(100)
-class With150MHz extends WithFrequency(150)
+// class With25MHz  extends WithFrequency(25)
+// class With50MHz  extends WithFrequency(50)
+// class With100MHz extends WithFrequency(100)
+// class With150MHz extends WithFrequency(150)
 
 class WithRocketCore extends Config(new WithNBigCores(1))
 class WithBOOMCore extends Config(new boom.common.WithNSmallBooms(1))
@@ -48,7 +48,7 @@ class StarshipBaseConfig extends Config(
   new BaseConfig().alter((site,here,up) => {
     case BootROMLocated(x) => up(BootROMLocated(x), site).map { p =>
       // invoke makefile for zero stage boot
-      val freqMHz = site(FPGAFrequencyKey).toInt * 1000000
+      val freqMHz = 100 * 1000000
       val path = System.getProperty("user.dir")
       val make = s"make -C firmware/zsbl ROOT_DIR=${path} img"
       println("[Leaving rocketchip] " + make)
diff --git a/repo/starship/src/main/scala/Top.scala b/repo/starship/src/main/scala/Top.scala
index c3db1cb..8b1bdde 100644
--- a/repo/starship/src/main/scala/Top.scala
+++ b/repo/starship/src/main/scala/Top.scala
@@ -11,10 +11,10 @@ import freechips.rocketchip.subsystem._
 import freechips.rocketchip.diplomacy._
 import freechips.rocketchip.devices.debug._
 import freechips.rocketchip.devices.tilelink._
-import sifive.fpgashells.devices.xilinx.xilinxvc707mig._
 
-import sifive.blocks.devices.spi._
-import sifive.blocks.devices.uart._
+// import sifive.fpgashells.devices.xilinx.xilinxvc707mig._
+// import sifive.blocks.devices.spi._
+// import sifive.blocks.devices.uart._
 
 class StarshipSystem(implicit p: Parameters) extends RocketSubsystem
   with HasAsyncExtInterrupts
diff --git a/repo/starship/src/main/scala/fpga/Configs.scala b/repo/starship/src/main/scala/fpga/Configs.scala
deleted file mode 100644
index c17bb64..0000000
--- a/repo/starship/src/main/scala/fpga/Configs.scala
+++ /dev/null
@@ -1,64 +0,0 @@
-package starship.fpga
-
-import starship._
-
-import chisel3._
-
-import freechips.rocketchip.system._
-import org.chipsalliance.cde.config._
-import freechips.rocketchip.subsystem._
-import freechips.rocketchip.diplomacy._
-import freechips.rocketchip.devices.debug._
-import freechips.rocketchip.devices.tilelink._
-
-import sifive.fpgashells.shell._
-import sifive.fpgashells.clocks._
-import sifive.fpgashells.ip.xilinx._
-import sifive.fpgashells.shell.xilinx._
-import sifive.fpgashells.devices.xilinx.xilinxvc707mig._
-
-import sifive.blocks.devices.spi._
-import sifive.blocks.devices.uart._
-
-
-case object VCU707DDRSizeKey extends Field[BigInt](0x40000000L) // 1 GB
-
-class WithPeripherals extends Config((site, here, up) => {
-  case PeripheryUARTKey => List(
-    UARTParams(address = BigInt(0x64000000L)))
-  case PeripherySPIKey => List(
-    SPIParams(rAddress = BigInt(0x64001000L)))
-  case MaskROMLocated(x) => List(
-    MaskROMParams(BigInt(0x20000L), "StarshipROM")
-  )
-})
-
-class StarshipFPGAConfig extends Config(
-  new WithPeripherals ++
-  new StarshipBaseConfig().alter((site,here,up) => {
-    case DebugModuleKey => None
-    case PeripheryBusKey => up(PeripheryBusKey, site).copy(dtsFrequency = Some(site(FrequencyKey).toInt * 1000000))
-    /* timebase-frequency = 1 MHz */
-    case DTSTimebase => BigInt(1000000L)
-    /* memory-size = 1 GB */
-    case MemoryXilinxDDRKey => XilinxVC707MIGParams(address = Seq(AddressSet(0x80000000L,site(VCU707DDRSizeKey)-1)))
-    case ExtMem => up(ExtMem, site).map(x => 
-      x.copy(master = x.master.copy(size = site(VCU707DDRSizeKey))))
-  })
-)
-
-class StarshipFPGADebugConfig extends Config(
-  new WithPeripherals ++
-  new WithJtagDTM ++
-  new WithClockGateModel() ++
-  new StarshipBaseConfig().alter((site,here,up) => {
-    //case DebugModuleKey => None
-    case PeripheryBusKey => up(PeripheryBusKey, site).copy(dtsFrequency = Some(site(FrequencyKey).toInt * 1000000))
-    /* timebase-frequency = 1 MHz */
-    case DTSTimebase => BigInt(1000000L)
-    /* memory-size = 1 GB */
-    case MemoryXilinxDDRKey => XilinxVC707MIGParams(address = Seq(AddressSet(0x80000000L,site(VCU707DDRSizeKey)-1)))
-    case ExtMem => up(ExtMem, site).map(x => 
-      x.copy(master = x.master.copy(size = site(VCU707DDRSizeKey))))
-  })
-)
\ No newline at end of file
diff --git a/repo/starship/src/main/scala/fpga/VC707.scala b/repo/starship/src/main/scala/fpga/VC707.scala
deleted file mode 100644
index 422db15..0000000
--- a/repo/starship/src/main/scala/fpga/VC707.scala
+++ /dev/null
@@ -1,87 +0,0 @@
-package starship.fpga
-
-import starship._
-import chisel3._
-import freechips.rocketchip.tile._
-import freechips.rocketchip.util._
-import freechips.rocketchip.prci._
-import org.chipsalliance.cde.config._
-import freechips.rocketchip.system._
-import freechips.rocketchip.tilelink._
-import freechips.rocketchip.subsystem._
-import freechips.rocketchip.diplomacy._
-import freechips.rocketchip.devices.debug._
-import freechips.rocketchip.devices.tilelink._
-import sifive.fpgashells.shell._
-import sifive.fpgashells.clocks._
-import sifive.blocks.devices.spi._
-import sifive.blocks.devices.uart._
-import sifive.fpgashells.ip.xilinx._
-import sifive.fpgashells.shell.xilinx.vc707shell._
-import sifive.fpgashells.devices.xilinx.xilinxvc707mig._
-
-
-class StarshipFPGATop(implicit p: Parameters) extends StarshipSystem
-  with HasPeripheryUART
-  with HasPeripherySPI
-  with HasPeripheryDebug
-  with HasMemoryXilinxVC707MIG
-{
-
-  val chosen = new DeviceSnippet {
-    def describe() = Description("chosen", Map(
-      "bootargs" -> Seq(ResourceString("nokaslr"))
-    ))
-  }
-  
-  val mmc = new MMCDevice(tlSpiNodes.head.device) 
-  ResourceBinding {
-    Resource(mmc, "reg").bind(ResourceAddress(0))
-  }
-
-  override lazy val module = new StarshipFPGATopModuleImp(this)
-
-}
-
-class StarshipFPGATopModuleImp[+L <: StarshipFPGATop](_outer: L) extends StarshipSystemModuleImp(_outer)
-  with HasPeripheryUARTModuleImp
-  with HasPeripherySPIModuleImp
-  with HasMemoryXilinxVC707MIGModuleImp
-  with DontTouch
-
-class TestHarness(override implicit val p: Parameters) extends VC707Shell
-    with HasDDR3 
-    with HasDebugJTAG 
-    {
-
-  dut_clock := (p(FPGAFrequencyKey) match {
-    case 25 => clk25
-    case 50 => clk50
-    case 100 => clk100
-    case 150 => clk150
-  })
-
-  withClockAndReset(dut_clock, dut_reset) {
-    val top = LazyModule(new StarshipFPGATop)
-    val dut = Module(top.module)
-    
-    connectSPI      (dut)
-    connectUART     (dut)
-    connectMIG      (dut)
-    connectDebugJTAG(top)
-
-    val childReset = (dut_reset.asBool | top.debug.map { debug => AsyncResetReg(debug.ndreset) }.getOrElse(false.B)).asBool
-    dut.reset := childReset
-
-    led := VecInit(0.U(8.W).asBools)
-    dut_ndreset := 0.U
-
-    dut.tieOffInterrupts()
-    dut.dontTouchPorts()
-
-    top.resetctrl.foreach { rc =>
-      rc.hartIsInReset.foreach { _ := childReset }
-    }
-    Debug.connectDebugClockAndReset(top.debug, dut_clock)
-  }
-}
\ No newline at end of file
